indirect.h

```diff
@@ -57,7 +67,7 @@ template <class T, class A>
 inline constexpr bool is_indirect_v<indirect<T, A>> = true;
 
 template <class T, class A = std::allocator<T>>
-class indirect {
+class XYZ_TRIVIALLY_RELOCATABLE indirect {
```

Comment:
This shouldn't be unconditional; it depends on the allocator. So the building block above should be
```
#define XYZ_TRIVIALLY_RELOCATABLE_IF(x) [[trivially_relocatable(x)]]
```
and then here should be either something simple-and-conservative:
```
class XYZ_TRIVIALLY_RELOCATABLE_IF((std::is_same_v<A, std::allocator<T>>)) indirect {
```
or else something like libc++'s implementation of `vector`:
```
template <class A>
struct allocator_pocma_models_relocatable : std::bool_constant<
    allocator_traits<A>::is_always_equal::value ||
    (allocator_traits<A>::propagate_on_container_copy_assignment::value &&
     allocator_traits<A>::propagate_on_container_move_assignment::value &&
     allocator_traits<A>::propagate_on_container_swap::value)
> {};

template <class A>
inline constexpr bool indirect_be_trivially_relocatable_v =
    std::is_trivially_relocatable_v<A> &&
    std::is_trivially_relocatable_v<typename allocator_traits<A>::pointer> &&
    allocator_pocma_models_relocatable<A>::value;

template<class T, class A = std::allocator<T>>
class XYZ_TRIVIALLY_RELOCATABLE_IF(indirect_be_trivially_relocatable_v<A>) indirect {
```

Reply:
Btw, to explain my naming choice here: Because of the precedent set by `<type_traits>`, I think of `foo_is_bar` as _asking a question_ — therefore I needed a different convention for _making an order_. Hence `foo_be_bar` (or sometimes `foo_should_be_bar`, but that's a longer identifier): read "Foo, _be_ bar!" So `foo_be_bar` determines the Platonic barness of foo, which in turn determines the value of `foo_is_bar`.
Not that I'm insisting you should adopt this convention; just explaining in case it looked like a typo. ;)